### PR TITLE
socketpair: omit the function definition on Windows for disable-socketpair

### PR DESCRIPTION
**Pull request by panjf2000**
Updates #13874

Ref: https://curl.se/dev/log.cgi?id=20240605035856-3529577

### CODE REVIEW

File: lib/socketpair.c [hunk 1]
```diff
@@ -239,6 +239,7 @@ int Curl_socketpair(int domain, int type, int protocol,
   return -1;
 }
 #else
+#ifndef _WIN32
```
Comment (MarcelRaad) [inline lib/socketpair.c:242]: Shouldn't this code block be dependent on `defined(HAVE_SOCKETPAIR) && !defined(CURL_DISABLE_SOCKETPAIR)` instead?

I think `HAVE_SOCKETPAIR` must be true as `socketpair` is used here, and `Curl_socketpair` must only be defined if `CURL_DISABLE_SOCKETPAIR` is not defined.

Comment (panjf2000) [reply lib/socketpair.c]: You're right! I checked out the code elsewhere, it seemed pointless to define this function with `CURL_DISABLE_SOCKETPAIR` specified.

**Review by MarcelRaad — APPROVED** [review]
Looks good to me and works for me, thanks! 👍

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**